### PR TITLE
*: Fix issue #7550 - Check correct error when opening rule files in tools

### LINKED ISSUE #7550: Check correct error when opening rule files in tools
**Thanos, Prometheus and Golang version used**:
go version go1.22.3 darwin/arm64

**Object Storage Provider**:

**What happened**:
There is a bug in this part of the code. If there is an issue in opening the file through `os.Open()`, then we will not be able to get exact error logs for debugging.
https://github.com/thanos-io/thanos/blob/35c0dbec856f97683a846e9c53f83156a3a44ff3/cmd/thanos/tools.go#L64-L70

**What you expected to happen**:
It should return error logs in case there is a issue in `os.Open()`.

**How to reproduce it (as minimally and precisely as possible)**:

**Full logs to relevant components**:

**Anything else we need to know**:
If you approve this issue, I will fix this issue and also propose full test coverage for tools.go .

### PR DESCRIPTION
**Pull request by NishantBansal2003**
* [ ] I added CHANGELOG entry for this change.
* [x] Change is not relevant to the end user.

## Changes
Fixes #7550 
There is a bug in this part of the code. If there is an issue with opening the file through `os.Open()`, we will not be able to obtain exact error logs for debugging. It should return error logs in case there is an issue with `os.Open()`. 
So, I fixed this bug and added test coverage.

## Verification
**Run the Tests**: Execute the following command to run the tests and verify that the bug has been fixed and the new tests are functioning correctly:
~~~
go test -v ./cmd/thanos
~~~
**Check Test Results**: Ensure that all tests pass and that there are no error messages related to the issue with os.Open()

### CODE REVIEW

**Comment by NishantBansal2003** [comment]
Hey, Please help me how to debug the failing checks.

**Comment by NishantBansal2003** [comment]
Hey @saswatamcode, Could you please review this PR and suggest potential improvements? Thank you!

File: cmd/thanos/tools.go [hunk 1]
```diff
@@ -65,7 +65,7 @@ func checkRulesFiles(logger log.Logger, patterns *[]string) error {
 			if er != nil {
 				level.Error(logger).Log("result", "FAILED", "error", er)
 				level.Info(logger).Log()
```
Comment (saswatamcode) [inline cmd/thanos/tools.go:67]: I think we can remove empty log as well?

**Review by saswatamcode — APPROVED** [review]
Thanks! 
But I think PR and issue title are a bit misleading. Maybe should be something like, "Check correct error when opening rule files in tools"?

File: cmd/thanos/tools.go [hunk 2]
```diff
@@ -55,7 +55,6 @@ func checkRulesFiles(logger log.Logger, patterns *[]string) error {
 		if err != nil || matches == nil {
 			err = errors.New("matching file not found")
 			level.Error(logger).Log("result", "FAILED", "error", err)
-			level.Info(logger).Log()
```
Comment (yeya24) [inline cmd/thanos/tools.go:58]: Do we need to remove this?

File: cmd/thanos/tools.go [hunk 3]
```diff
@@ -64,8 +63,7 @@ func checkRulesFiles(logger log.Logger, patterns *[]string) error {
 			f, er := os.Open(fn)
 			if er != nil {
 				level.Error(logger).Log("result", "FAILED", "error", er)
-				level.Info(logger).Log()
-				failed.Add(err)
+				failed.Add(er)
```
Comment (yeya24) [inline cmd/thanos/tools.go:66]: Good catch

File: cmd/thanos/tools.go (hunk 2 above)
Comment (NishantBansal2003) [reply cmd/thanos/tools.go:58]: These are just empty logs, Its output is - `level=info`, So I removed all `level.Info(logger).Log()`

Comment (yeya24) [reply cmd/thanos/tools.go:58]: I think it might be intended to leave some spaces between lines.

Comment (NishantBansal2003) [reply cmd/thanos/tools.go:58]: Ok, Should I reintroduce all instances of `level.Info(logger).Log()` throughout the entire code, or only in this specific part of the code?

Comment (yeya24) [reply cmd/thanos/tools.go:58]: I don't have strong preference. I am ok either way to keep them or remove them.

**Comment by NishantBansal2003** [comment]
Hey @saswatamcode,
I removed all instances of empty logs (`level.Info(logger).Log()`) in the tools. However, if there's a need to bring back any specific ones, please let me know.

**Review by saswatamcode — APPROVED** [review]

**Pull request merged**